### PR TITLE
add LFMMI loss

### PR DESCRIPTION
**Pull request by aluminumbox**
add LFMMI loss, add torch.jit.ignore for lfmmi function

### CODE REVIEW

File: wenet/transformer/asr_model.py [hunk 1]
```diff
@@ -89,6 +94,9 @@ def forward(
             text: (Batch, Length)
             text_lengths: (Batch,)
         """
+        if self.lfmmi_dir != '':
```
Comment (robin1001) [inline wenet/transformer/asr_model.py:97]: Why load it in forward? I think we should load it in construction.

Comment (robin1001) [reply wenet/transformer/asr_model.py:97]: Then there is no need to use hasattr when loading resource

File: wenet/transformer/asr_model.py [hunk 2]
```diff
@@ -552,6 +563,54 @@ def attention_rescoring(
...
+            with open('{}/words.txt'.format(self.lfmmi_dir), 'r') as fin:
+                for line in fin:
+                    arr = line.strip().split()
+                    assert len(arr) == 2
+                    self.word_table[int(arr[1])] = arr[0]
+
+    @torch.jit.ignore(drop=True)
+    def _calc_lfmmi_loss(self, encoder_out, encoder_mask, text):
+        ctc_probs = self.ctc.log_softmax(encoder_out)
+        supervision_segments = torch.stack(
```
Comment (robin1001) [inline wenet/transformer/asr_model.py:599]: why we should move it to cpu?

File: wenet/transformer/asr_model.py (hunk 1 above)
Comment (aluminumbox) [reply wenet/transformer/asr_model.py:97]: > Why load it in forward? I think we should load it in construction.

because i need to decorate it with torch.jit.ignore, i dont want to decorate whole forward function with torch.jit.ignore

File: wenet/transformer/asr_model.py (hunk 2 above)
Comment (aluminumbox) [reply wenet/transformer/asr_model.py:599]: k2 supervision_segments  on cpu

File: wenet/transformer/asr_model.py (hunk 1 above)
Comment (aluminumbox) [reply wenet/transformer/asr_model.py]: now we move load_mmi_resource into __init__ and remove hasattr judgement

**Review by robin1001 — APPROVED** [review]

**Comment by kli017** [comment]
@aluminumbox 请问现在lfmmi在runtime推理中能够使用吗
